File: returned/python/today13/min_stack.py
class MinStack:
    def __init__(self):
        self.stack = [(None, float('inf'))]

    def push(self, val: int) -> None:
        nextMin = min(val, self.stack[-1][1])
        self.stack.append((val, nextMin))

    def pop(self) -> None:
        self.stack.pop()

    def top(self) -> int:
        return self.stack[-1][0]

    def getMin(self) -> int:
        return self.stack[-1][1]

File: csharp/Leet/Today1/MinStack.cs
// https://leetcode.com/problems/min-stack/description/

namespace Today1.MinStack;

public class MinStack
{
    Stack<(int Value, int Minimum)> stack = new Stack<(int, int)>();

    public MinStack()
    {
        stack.Push((0, int.MaxValue));
    }

    public void Push(int val)
    {
        var min = Math.Min(val, stack.Peek().Minimum);
        stack.Push((val, min));
    }

    public void Pop()
    {
        stack.Pop();
    }

    public int Top()
    {
        return stack.Peek().Value;
    }

    public int GetMin()
    {
        return stack.Peek().Minimum;
    }
}